packages/js/src/utils/base-module.ts

```diff
@@ -0,0 +1,48 @@
+import { ApiService } from 'client/dist/cjs';
+import { NovuEventEmitter } from '../event-emitter';
+
+interface CallQueueItem {
+  fn: () => Promise<unknown>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  resolve: (value: any | PromiseLike<any>) => void;
+  reject: (reason?: unknown) => void;
+}
+
+export class BaseModule {
+  _apiService: ApiService;
+  _emitter: NovuEventEmitter;
+  #callsQueue: CallQueueItem[] = [];
+  #sessionError: unknown;
+
+  constructor(emitter: NovuEventEmitter, apiService: ApiService) {
+    this._emitter = emitter;
+    this._apiService = apiService;
+    this._emitter.on('session.initialize.success', () => {
+      this.#callsQueue.forEach(async ({ fn, resolve }) => {
+        resolve(await fn());
+      });
+      this.#callsQueue = [];
+    });
+    this._emitter.on('session.initialize.error', ({ error }) => {
+      this.#sessionError = error;
+      this.#callsQueue.forEach(({ reject }) => {
+        reject(error);
+      });
+      this.#callsQueue = [];
+    });
```

Comment:
when the session initialization has failed reject all pending promises with that error